### PR TITLE
LP-2912 Fix webinar invitations/update emails

### PR DESCRIPTION
**Pull request by asadali145**
[LP-2912](https://philanthropyu.atlassian.net/browse/LP-2912)

***

#### Description:

- Send invites to only new members added in webinar while updating it.
- Do not send update emails to newly added members while updating a webinar.

### CODE REVIEW

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3161/workflows/workflows_id/jobs/3161/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/webinars/helpers.py (85.7%): Missing lines 304
openedx/adg/lms/webinars/models.py (100%)
openedx/adg/lms/webinars/tests/test_helpers.py (100%)
-------------
Total:   9 lines
Missing: 1 line
Coverage: 88%
-------------
```

File: openedx/adg/lms/webinars/helpers.py [hunk 1]
```diff
@@ -277,23 +277,32 @@ def send_webinar_invitation_emails(webinar, old_webinar_data=None):
     if invite_email_recipients:
         send_webinar_emails(MandrillClient.WEBINAR_CREATED, webinar, list(invite_email_recipients))
 
-    if unregistered_members_emails:
-        send_webinar_emails(MandrillClient.WEBINAR_TEAM_INVITE, webinar, list(unregistered_members_emails))
+    member_email_recipients_with_no_platform_account = unregistered_members_emails.intersection(updated_webinar_emails)
```
Comment (aightmunam) [inline openedx/adg/lms/webinars/helpers.py:280]: Can we use `&` for set intersection to make it consistent with the other code. We are using `-` for set difference.

File: openedx/adg/lms/webinars/helpers.py [hunk 2]
```diff
@@ -277,23 +277,32 @@ def send_webinar_invitation_emails(webinar, old_webinar_data=None):
...
     Args:
+        old_webinar_data (dict): Dict containing old data for webinar
         webinar (Webinar): The webinar to send update emails for
     """
+    old_webinar_emails = get_webinar_recipient_emails(old_webinar_data) if old_webinar_data else None
+
     webinar_team_emails = webinar.team_emails
     user_emails_registered_in_webinar = set(webinar.get_webinar_update_recipients_emails())
 
     emails_to_send_webinar_updates = user_emails_registered_in_webinar.union(webinar_team_emails)
```
Comment (aightmunam) [inline openedx/adg/lms/webinars/helpers.py:301]: Similarly, let's use `|` for union.

File: openedx/adg/lms/webinars/helpers.py [hunk 3]
```diff
@@ -277,23 +277,32 @@ def send_webinar_invitation_emails(webinar, old_webinar_data=None):
...
 
+    if old_webinar_emails:
+        emails_to_send_webinar_updates = emails_to_send_webinar_updates.intersection(old_webinar_emails)
+
     send_webinar_emails(MandrillClient.WEBINAR_UPDATED, webinar, list(emails_to_send_webinar_updates))
```
Comment (aightmunam) [inline openedx/adg/lms/webinars/helpers.py:306]: Also, haven't we decided to send webinar invitation emails i.e WEBINAR_CREATED template, to all the invitees that aren't yet registered in the webinar?

File: openedx/adg/lms/webinars/helpers.py [hunk 4]
```diff
@@ -277,23 +277,32 @@ def send_webinar_invitation_emails(webinar, old_webinar_data=None):
...
     """
+    old_webinar_emails = get_webinar_recipient_emails(old_webinar_data) if old_webinar_data else None
+
     webinar_team_emails = webinar.team_emails
     user_emails_registered_in_webinar = set(webinar.get_webinar_update_recipients_emails())
 
     emails_to_send_webinar_updates = user_emails_registered_in_webinar.union(webinar_team_emails)
 
+    if old_webinar_emails:
+        emails_to_send_webinar_updates = emails_to_send_webinar_updates.intersection(old_webinar_emails)
```
Comment (aightmunam) [inline openedx/adg/lms/webinars/helpers.py:304]: Can we add a test that covers different scenarios here?

**Review by aightmunam — CHANGES_REQUESTED** [review]

File: openedx/adg/lms/webinars/helpers.py (hunk 3 above)
Comment (aightmunam) [reply openedx/adg/lms/webinars/helpers.py:306]: Oh that will be covered in the invite emails function, I believe.

File: openedx/adg/lms/webinars/helpers.py (hunk 1 above)
Comment (asadali145) [reply openedx/adg/lms/webinars/helpers.py:280]: I think `intersection` is more clear and readable as compared to `&`. WDYT?

File: openedx/adg/lms/webinars/helpers.py (hunk 2 above)
Comment (asadali145) [reply openedx/adg/lms/webinars/helpers.py:301]: Same here, I think union is better for readability.

File: openedx/adg/lms/webinars/helpers.py (hunk 1 above)
Comment (aightmunam) [reply openedx/adg/lms/webinars/helpers.py:280]: Then let's use `difference` instead of `-` to make it consistent everywhere.

Comment (annas-usman) [inline openedx/adg/lms/webinars/helpers.py:280]: I think `member_email_recipients_with_no_platform_account` is too long of a name for a variable, maybe we can rename it to something like `new_unregistered_members_emails` as we are extracting those emails that have been newly added and there is no platform account against those emails.

**Review by annas-usman — CHANGES_REQUESTED** [review]

**Review by aightmunam — APPROVED** [review]
LGTM!

**Review by annas-usman — APPROVED** [review]

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3168/workflows/workflows_id/jobs/3168/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/webinars/helpers.py (88.9%): Missing lines 304
openedx/adg/lms/webinars/models.py (100%)
openedx/adg/lms/webinars/tests/test_helpers.py (100%)
-------------
Total:   17 lines
Missing: 1 line
Coverage: 94%
-------------
```

**Pull request merged**